api/src/opentelemetry/trace/span_id.cc

```diff
@@ -0,0 +1,29 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <opentelemetry/trace/span_id.h>
+
+#include <cstdint>
+#include <cstring>
+#include <string>
+
+namespace opentelemetry {
+namespace trace {
+
+SpanId::SpanId(const uint8_t *buf) { memcpy(rep_, buf, kSize); }
+
+std::string SpanId::ToHex() const { return "unimplemented"; }
```

Comment:
I would expect a way to format SpanId without having to dynamically allocating memory.
For example, in a critical service I want to preallocate the buffer for SpanId hex format, so I could dump the trace_id/span_id to stderr under critical situation, without having to create `std::string` on the fly (double fault or triple fault).

Reply:
I agree. Maybe we can add a back port of [std::span](https://en.cppreference.com/w/cpp/container/span) for this? or some other version that takes a pointer to a buffer

Reply:
I agree with having a fast version that we can point at a pre-allocated buffer. This can be added later. It's harder to take away from an API. :)

Reply:
Resolved by implementing fast version.